perception/object_recognition/detection/shape_estimation/CMakeLists.txt

```diff
@@ -1,36 +1,25 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(shape_estimation)
 
-add_compile_options(-std=c++14 -O3)
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  autoware_perception_msgs
-  pcl_ros
-  roscpp
-  tf2
-  tf2_geometry_msgs
-  rostest
-  rosunit
-  eigen_conversions
-)
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(autoware_perception_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+find_package(tf2_eigen REQUIRED)
+
+find_package(PCL REQUIRED)
+find_package(pcl_conversions REQUIRED)
 
 find_package(OpenCV REQUIRED)
 find_package(Eigen3 REQUIRED)
 
-catkin_package(
-  INCLUDE_DIRS
-    include
-  CATKIN_DEPENDS
-    autoware_perception_msgs
-    pcl_ros
-    roscpp
-    tf2
-    tf2_geometry_msgs
-    eigen_conversions
-)
 
 include_directories(include
```

Comment:
We should be able to handle the OpenCV and Eigen includes in `ament_target_dependencies`, by adding `Eigen3` to `SHAPE_ESTIMATION_DEPENDENCIES`.

Instead of `include_directories(include)`, you can then write
```
target_include_directories(JSONUtils
  PUBLIC 
    $<INSTALL_INTERFACE:include>    
    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
  PRIVATE
    ${CMAKE_CURRENT_SOURCE_DIR}/src
)
```
See https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right/ – we have headers in `src/` as well which I assume should not be needed by downstream packages, so the `${CMAKE_CURRENT_SOURCE_DIR}/src` is theoretically needed. In practice it seems to compile fine when I leave it out, not sure why, but it's better to keep it in.

Reply:
Done.